services-examples-parent/scalecube-examples-spring/src/main/java/io/scalecube/spring/SpringServicesProvider.java

```diff
@@ -0,0 +1,103 @@
+package io.scalecube.spring;
+
+import io.scalecube.services.Microservices;
+import io.scalecube.services.ServiceCall;
+import io.scalecube.services.ServiceDefinition;
+import io.scalecube.services.ServiceInfo;
+import io.scalecube.services.ServicesProvider;
+import io.scalecube.services.annotations.Service;
+import io.scalecube.services.discovery.api.ServiceDiscovery;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.support.GenericBeanDefinition;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import org.springframework.util.ClassUtils;
+import reactor.core.publisher.Mono;
+
+public class SpringServicesProvider implements ServicesProvider {
+
+  private final Class<?>[] configurations;
+
+  // lazy init
+  private AnnotationConfigApplicationContext ctx;
+
+  /**
+   * Create the instance from Spring Context.
+   *
+   * @param configurations spring configurations.
+   */
+  public SpringServicesProvider(Class<?>... configurations) {
+    this.configurations = configurations;
+  }
+
+  @Override
+  public Mono<Collection<ServiceDefinition>> provideServiceDefinitions(
+      Microservices microservices) {
+
+    this.ctx = new AnnotationConfigApplicationContext();
```

Comment:
Move it to constructor and make it as final field.

Reply:
This is only a temporary example for internal use. Before I merged this pull request, I was supposed to delete it. Since this is just a demonstration of how to use Spring IoC

Reply:
> This is only a temporary example for internal use. Before I merged this pull request, I was supposed to delete it. Since this is just a demonstration of how to use Spring IoC

Whole idea is to create an example of **working** custom DI mechanism. Not a partial support not a half-working solution, but holistic one. So if u're going to delete exactly this one, then create another ones to how: how to use it with a simple standalone spring, how to use in the context of springboot.